web/web_types/types.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package web_types
+
+import (
+	"net/url"
+	"time"
+)
+
+// RunRuntimeInfo contains runtime info about Prometheus.
+type RuntimeInfo struct {
+	Birth               time.Time          `json:"startTime"`
+	CWD                 string             `json:"cwd"`
+	Version             *PrometheusVersion `json:"version"`
+	Alertmanagers       []*url.URL         `json:"alertmanagers"`
```

Comment:
I think it should be a slice of strings instead of `url.URL`. This is the current API output with 1 configured AlertManager:

```
{
  "status": "success",
  "data": {
...
    "alertmanagers": [
      {
        "Scheme": "http",
        "Opaque": "",
        "User": null,
        "Host": "localhost:9093",
        "Path": "/api/v1/alerts",
        "RawPath": "",
        "ForceQuery": false,
        "RawQuery": "",
        "Fragment": ""
      }
    ],
...
  }
}
```

Reply:
@simonpasquier The `status.html` template uses the `Host`, `Path` and `Scheme`.

https://github.com/prometheus/prometheus/blob/4a0b6aa612daf141b86c49e9207542eea1018ebd/web/ui/templates/status.html#L90

I'm not sure what to do. Any help will be appreciated.

Reply:
Right, I would be inclined to declare an intermediary `Alertmanager` struct that would only expose the required fields.

```go
type AlertManager struct {
    Scheme string `json:"scheme"`
    Host string `json:"host"`
    Path string `json:"path"`
}

type RuntimeInfo struct {
...
    Alertmanagers []Alertmanager `json:"alertmanagers"`
...
}
```